plugins/database/influxdb/influxdb.go

```diff
@@ -129,21 +129,25 @@ func (i *Influxdb) CreateUser(ctx context.Context, statements dbplugin.Statement
 				"password": password,
 			}), "", "")
 			response, err := cli.Query(q)
-			if err != nil && response.Error() != nil {
-				for _, stmt := range rollbackIFQL {
-					for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
-						query = strings.TrimSpace(query)
-
-						if len(query) == 0 {
-							continue
-						}
-						q := influx.NewQuery(dbutil.QueryHelper(query, map[string]string{
-							"username": username,
-						}), "", "")
-
-						response, err := cli.Query(q)
-						if err != nil && response.Error() != nil {
-							return "", "", err
+			if err != nil {
+				if response != nil && response.Error() != nil {
+					for _, stmt := range rollbackIFQL {
+						for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+							query = strings.TrimSpace(query)
+
+							if len(query) == 0 {
+								continue
+							}
+							q := influx.NewQuery(dbutil.QueryHelper(query, map[string]string{
+								"username": username,
+							}), "", "")
+
+							response, err := cli.Query(q)
+							if err != nil {
+								if response != nil && response.Error() != nil {
+									return "", "", err
```

Comment:
We chatted about this out-of-band and agreed that although it seems strange that the code only returns in one error condition and not the other, we don't want to make any logical changes here because the only goal is to resolve the panic.

Reply:
👍 thanks 😄